### ISSUE TITLE
eCAL 5.10 broke compatibility with eCAL 5.9 and lower in C API

### ISSUE THREAD

**Issue by FlorianReimold** [continental/ecal#671]
Some functions were renamed without any need to rename them.

Commit:
https://github.com/continental/ecal/commit/17f576bc749461be597de5afb1993102593fe102#diff-d689389f1a48931b2e6f5dce34cff0ff5b8ac0a8efe2a99879faca9b01742a67

Files (probably more):
ecal/core/include/ecal/cimpl/ecal_proto_dyn_json_subscriber_cimpl.h
ecal/core/include/ecal/cimpl/ecal_client_cimpl.h
ecal/core/include/ecal/cimpl/ecal_publisher_cimpl.h
ecal/core/include/ecal/cimpl/ecal_subscriber_cimpl.h

**Comment by rex-schilasky** [comment]
Yes, this was done by accident. The idea was to have the same naming convention for all C API functions. Some of them end with a 'C' some not. This C postfix makes no sense but we can rename then back for the minority of users that are using the raw C API.

**Comment by Kerstin-Keller** [comment]
I will make the fix. We will have both and deprecate the old `C` postfix. We can then remove the deprecated functions with eCAL 6.

The deprecation may be compiler specific, and might not work on all platforms.
However it will be marked as deprecated in the documentation.

Should we wait with the deprecation for 5.11. and have both (none deprecated) in 5.10? Because then we can mention it in the release notes.

**Comment by FlorianReimold** [comment]
The deprecation notice is not really consistent anyways. But please add some "Deperecated since 5.11" or whatever to the documentation, so we can add more consistent deprecation notes to the release notes in the future
